google-cloud-logging/lib/google/cloud/logging/async_writer.rb

```diff
@@ -459,40 +461,34 @@ class Batch
 
           def initialize writer
             @writer = writer
-            @entries = {}
-            @total_message_bytes = 0
+            @entries = Concurrent::Array.new
```

Comment:
I don't think we need `Concurrent::Array`. The array is written only from `write_entries` which is synchronized, and read only from a single worker thread, and the two phases never overlap.